File: precourse_javascript/05_flow_control/ex_08.js
// Write a function that logs whether a number is between 0 and 50(inclusive), between 51 and 100(inclusive), greater than 100, or less than 0.

function numberRange(num) {
  if (num > 0 && num <= 50) {
    console.log(`${num} is between 0 and 50`);
  } else if (num > 50 && num < 100) {
    console.log(`${num} is between 50 and 100`);
  } else if (num > 100) {
    console.log(`${num} is greater than 100`);
  } else {
    console.log(`${num} is less than 0`);
  }
}

numberRange(25);   // '25 is between 0 and 50'
numberRange(75);   // '75 is between 51 and 100'
numberRange(125);  // '125 is greater than 100'
numberRange(-25);  // '-25 is less than 0'

File: RB101/exercises/02_easy/ex_08.rb
def calculate_sum(num)
  (1..num).to_a.sum
end

def calculate_product(num)
  result = 1

  (1..num).to_a.each  { |n| result *= n }

  result
end

puts '>> Please enter an integer greater than 0:'
number = gets.to_i

puts ">> Enter 's' to compute the sum, 'p' to compute the product."
operation = gets.chomp

if operation == 's'
  sum = calculate_sum(number)
  puts "The sum of the integers between 1 and #{number} is #{sum}."
else
  product = calculate_product(number)
  puts "The product of the integers between 1 and #{number} is #{product}."
end

# >> Please enter an integer greater than 0:
# 5
# >> Enter 's' to compute the sum, 'p' to compute the product.
# s
# The sum of the integers between 1 and 5 is 15.


# >> Please enter an integer greater than 0:
# 6
# >> Enter 's' to compute the sum, 'p' to compute the product.
# p
# The product of the integers between 1 and 6 is 720.
